aioxmpp/xso/model.py

```diff
@@ -1479,10 +1479,23 @@ class ChildTextMap(ChildValueMap):
     convert the :class:`AbstractTextChild` subclass `xso_type` to and from
     a language-text mapping.
 
+    If instead of an :class:`XSO` a tag is passed (that is, a valid
+    argument to :func:`normalize_tag`) an :class:`AbstractTextChild`
+    instance is created on demand.
+
     For an example, see :class:`.Message`.
     """
 
     def __init__(self, xso_type):
+        if not isinstance(xso_type, XMLStreamClass):
+            tag = normalize_tag(xso_type)
+
+            xso_type = type(
+                "TextChild" + tag[1],
+                (AbstractTextChild,),
+                {"TAG": tag},
+            )
```

Comment:
Hmm, does this work? I’d expect that this complains about the meta class not matching (because ``AbstractTextChild`` has the ``XMLStreamClass`` metaclass, not ``type`` itself).

Reply:
I think it does, but it may be better to use the proper metaclass instead of type. I will investigate, but since the docs say this is the "dynamic form of class" <https://docs.python.org/3/library/functions.html#type> I would guess, that it correctly determines the metaclass from the bases (in which case this form would be resistant against renaming XMLStreamClass or using a subclass of XMLStreamClass as the metaclass of AbstractTextChild).

Reply:
Ok, agreed. Let’s keep it this way.